x-pack/plugins/observability_solution/inventory/public/components/entities_grid/index.tsx

```diff
@@ -64,6 +69,16 @@ export function EntitiesGrid({
     [onChangeSort]
   );
 
+  const onEntityNameClick = useCallback(
+    ({ entityType }: { entityType: EntityType }) => {
+      telemetry.reportEntityViewClicked({
+        view_type: 'detail',
+        entity_type: entityType,
+      });
+    },
+    [telemetry]
+  );
```

Comment:
I had the EntitiesGrid handle the click event because I envisioned scenarios where the EntityName component might be used in other contexts where we might not want to trigger this event.

If you think it doesn't make sense and we want the event to always be associated with EntityName, I can easily move it to be managed within the EntityName component instead.

Reply:
I can't see atm why and where `EntityName` could be reused outside the inventory grid. Aside from the link to the Detail Views, there isn't much there to make it reusable - but maybe I'm missing something.

Reply:
Perfect, no problem. I'll move it and keep it within `EntityName`.

https://github.com/elastic/kibana/pull/197055/commits/52f94cd782bd92974e49729a7c11d5f29c21d8ad